### PR TITLE
Fix TTL when config has not been set. 

### PR DESCRIPTION
**Pull request by emilymye**
Was panic on nil - adding reviewers for now since I'm not sure what the TTL changes were.

### CODE REVIEW

**Comment by chrishoffman** [comment]
This is not required as passing in the zero values will default to these values in core. Are you seeing an issue with incorrect values being set?

File: plugin/secrets_service_account_key.go [hunk 1]
```diff
@@ -99,7 +99,10 @@ func (b *backend) secretKeyRenew(ctx context.Context, req *logical.Request, d *f
 		return nil, err
 	}
 	if cfg == nil {
-		cfg = &config{}
+		cfg = &config{
```
Comment (chrishoffman) [inline plugin/secrets_service_account_key.go]: This change is not required.  You can just pass the zero values back and they will get set appropriately.

File: plugin/secrets_service_account_key.go [hunk 2]
```diff
@@ -170,6 +173,12 @@ func (b *backend) getSecretKey(ctx context.Context, s logical.Storage, rs *RoleS
 	if err != nil {
 		return nil, errwrap.Wrapf("could not read backend config: {{err}}", err)
 	}
+	if cfg == nil {
+		cfg = &config{
```
Comment (chrishoffman) [inline plugin/secrets_service_account_key.go]: You can just initialize an empty struct here.  No need to query the system values.

Comment (chrishoffman) [reply plugin/secrets_service_account_key.go]: I missed this panic on first pass and that was the reason for my initial comment.

**Comment by jefferai** [comment]
@emilymye The TTL changes were a part of us going through all backends and plugins and modifying them so that Vault's core, not each backend, is now in charge of actual TTL issue/renewal calculations. This is safer and more standardized but required some updating of code in most backends.

File: plugin/secrets_service_account_key.go (hunk 2 above)
Comment (jefferai) [reply plugin/secrets_service_account_key.go]: But, you still don't need to set the values :-)

**Comment by emilymye** [comment]
SG, removed the values - I'm going to go ahead and merge!

**Comment by jefferai** [comment]
👍

**Pull request merged**